e2e/app.po.ts

```diff
@@ -118,4 +137,23 @@ export class AppPage {
     browser.waitForAngularEnabled(true);
     return this.goToUrl(AppPage.baseurl);
   }
+
+
+  /**
+   * Hook into browser and fetch config.json
+   * @returns {any} config.json used in ipaas app
+   */
+  getSettings(): P<any> {
+    // jquery is invoked in the context of the browser
+    return browser.driver.executeAsyncScript((callback) => {
+      jQuery.get('/config.json', function (data) {
+        callback(data);
+      });
+    }).then(jsonString => JSON.parse(<string> jsonString));
```

Comment:
This is what I like the most on protractor - it allow me to easily invoke method in context of browser
and I still have syntax highlight and autocompletion here :)

It uses jquery callback inside browser and return obtained data as Promise.

Usage:
```ts
  async getApiUrl(): P<string> {
    const settings = await this.getSettings();
    return settings.apiEndpoint;
  }
```